dotnet/src/Connectors/Connectors.Memory.SqlServer/SqlServerCommandBuilder.cs

```diff
@@ -134,8 +136,13 @@ internal static SqlCommand MergeIntoSingle(
 
         foreach (var property in model.Properties)
         {
-            sb.AppendParameterName(property, ref paramIndex, out string paramName).Append(',');
-            command.AddParameter(property, paramName, record[property.StorageName]);
+            sb.AppendParameterName(property, ref paramIndex, out var paramName).Append(',');
+
+            var value = property is VectorPropertyModel vectorProperty && generatedEmbeddings?.TryGetValue(vectorProperty, out var ge) == true
+                ? ge[0]
```

Comment:
`ge` does not tell much. Also, why do we take only the first one?

```suggestion
            var value = property is VectorPropertyModel vectorProperty && generatedEmbeddings?.TryGetValue(vectorProperty, out var embeddings) == true
                ? embeddings[0]
```